web3/_utils/encoding.py

```diff
@@ -88,34 +104,34 @@ def to_hex_twos_compliment(value, bit_size):
 
     value = (1 << bit_size) + value
     hex_value = hex(value)
-    hex_value = hex_value.rstrip("L")
+    hex_value = HexStr(hex_value.rstrip("L"))
     return hex_value
 
 
-def to_hex_with_size(value, bit_size):
+def to_hex_with_size(value: Any, bit_size: int) -> HexStr:
     """
     Converts a value to hex with given bit_size:
     """
     return pad_hex(to_hex(value), bit_size)
 
 
-def pad_hex(value, bit_size):
+def pad_hex(value: Any, bit_size: int) -> HexStr:
     """
     Pads a hex string up to the given bit_size
     """
     value = remove_0x_prefix(value)
     return add_0x_prefix(value.zfill(int(bit_size / 4)))
 
 
-def trim_hex(hexstr):
+def trim_hex(hexstr: HexStr) -> HexStr:
     if hexstr.startswith('0x0'):
-        hexstr = re.sub('^0x0+', '0x', hexstr)
+        hexstr = HexStr(re.sub('^0x0+', '0x', hexstr))
```

Comment:
orthogonal to this PR but I'd be curious of the performance difference between this implementation and just converting to an integer and then back to hex...